From 17266b511cb2c15fe1592bbf82f04fdefeebb059 Mon Sep 17 00:00:00 2001
From: Tao Zeng <tao.zeng@amlogic.com>
Date: Mon, 18 Jan 2016 16:03:22 +0800
Subject: [PATCH 5936/5965] PD#118299: cec: imporve 32k clock resolution for
 m8baby costdown platform

32k clock of cec is generated by PWM on costdown platform.
Old clock config is divide down from 637.5MHz, and can only get:
32771.294916Hz with nearly 100ppm, which is not accurate enough.

Now we use 850MHz to divid down 32k, and we can get:
32767.925983Hz with only 3ppm, it is stable enough.

Change-Id: I74967778ebf01046b8125cc7c1735877933420da
---
 arch/arm/mach-meson8b/hdmi_tx_hw/hdmi_tx_cec_hw.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/arch/arm/mach-meson8b/hdmi_tx_hw/hdmi_tx_cec_hw.c b/arch/arm/mach-meson8b/hdmi_tx_hw/hdmi_tx_cec_hw.c
index 7eb479bf41fd..ea9baa247296 100755
--- a/arch/arm/mach-meson8b/hdmi_tx_hw/hdmi_tx_cec_hw.c
+++ b/arch/arm/mach-meson8b/hdmi_tx_hw/hdmi_tx_cec_hw.c
@@ -73,11 +73,11 @@ void gpiox_10_pin_mux_mask(void){
 
 void pwm_e_config(void){
     //PWM E config
-    aml_set_reg32_bits(P_PWM_PWM_E,    0x25ff, 16, 16); //0x21b0 bit[31:16]: PWM_E_HIGH counter
-    aml_set_reg32_bits(P_PWM_PWM_E,    0x25fe,  0, 16); //0x21b0 bit[15: 0]: PWM_E_LOW counter
+    aml_set_reg32_bits(P_PWM_PWM_E,    0x32aa, 16, 16); //0x21b0 bit[31:16]: PWM_E_HIGH counter
+    aml_set_reg32_bits(P_PWM_PWM_E,    0x32aa,  0, 16); //0x21b0 bit[15: 0]: PWM_E_LOW counter
     aml_set_reg32_bits(P_PWM_MISC_REG_EF, 0x1, 15,  1); //0x21b2 bit[15]   : PWM_E_CLK_EN
     aml_set_reg32_bits(P_PWM_MISC_REG_EF, 0x0,  8,  7); //0x21b2 bit[14: 8]: PWM_E_CLK_DIV
-    aml_set_reg32_bits(P_PWM_MISC_REG_EF, 0x2,  4,  2); //0x21b2 bit[5 : 4]: PWM_E_CLK_SEL :0x2 sleect fclk_div4:637.5M
+    aml_set_reg32_bits(P_PWM_MISC_REG_EF, 0x3,  4,  2); //0x21b2 bit[5 : 4]: PWM_E_CLK_SEL :0x2 sleect fclk_div3:850MHz
     aml_set_reg32_bits(P_PWM_MISC_REG_EF, 0x1,  0,  1); //0x21b2 bit[0]    : PWM_E_EN
 }
 
-- 
2.19.0

